### PR TITLE
Add function QueryVmList

### PR DESCRIPTION
**Pull request by dataclouder**
Add function `client.QueryVmList` to retrieve VMs without a VDC constraint, and with the additional possibility of applying custom filters

### CODE REVIEW

File: govcd/vm_affinity_rule_test.go [hunk 1]
```diff
@@ -271,58 +271,3 @@ func (vcd *TestVCD) Test_VmAffinityRule(check *C) {
 	}, check)
```
Comment (dataclouder) [inline govcd/vm_affinity_rule_test.go:272]: Note to reviewers: function moved to common_test.go

**Review by Didainius — APPROVED** [review]
Test passed - thanks!

**Review by adambarreiro — APPROVED** [review]

**Review by arunmk — APPROVED** [review]
Thanks for the change @dataclouder

File: govcd/vm.go [hunk 2]
```diff
@@ -1704,6 +1704,35 @@ func queryVmList(filter types.VmQueryFilter, client *Client, filterParent, filte
...
+	if vmType.String() != "" {
+		// The first filter will be the type of VM, i.e. deployed (inside a vApp) or not (inside a vApp template)
+		filterText = vmType.String()
+	}
+	for k, v := range searchParams {
+		filterText = fmt.Sprintf("%s;%s==%s", filterText, k, v)
+	}
+
+	params["filter"] = filterText
+	vmResult, err := client.cumulativeQuery(queryType, nil, params)
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply govcd/vm.go:1725]: The limit is the client available memory.
What is the maximum number of VMs you have had in a single Organization? Could you try it without filters?

**Review by lvirbalas — APPROVED** [review]

**Pull request merged**